pkg/apis/openebs.io/scp/v1alpha1/storage-pool.go

```diff
@@ -29,8 +31,387 @@ import (
+	MaxDeviceCount uint64 `json:"maxDeviceCount,omitempty"`
+
+	// DeviceTypeIdentifier for the device viz.  ssd-4k, ssd-16k, nvme-64k
+	DeviceTypeIdentifier StoragePoolDeviceTypeIdentifier `json:"deviceTypeIdentifier,omitempty"`
+
+	// DeviceSelector for labels to match with devices incase they are labeled
+	DeviceSelector map[string]string `json:"deviceSelector,omitempty"`
+
+	// DeviceSelector for filtering devices based on FiltersOptions
+	DeviceFilters map[string][]FilterOptions `json:"deviceFilters,omitempty"`
+
+	// Devices for the UIDs of the free devices
+	Devices []types.UID `json:"devices,omitempty"`
+}
+
+type StoragePoolCapabilities struct {
+	// DataStorageCapabilities for the storage pool
+	DataStorageCapabilities StoragePoolDataStorageCapabilities `json:"dataStorageCapabilities,omitempty"`
+
+	// DataSecurityCapabilities for the storage pool
+	DataSecurityCapabilities StoragePoolDataSecurityCapabilities `json:"dataSecurityCapabilities,omitempty"`
+
+	// IOConnectivityCapabilities for the storage pool
+	IOConnectivityCapabilities StoragePoolIOConnectivityCapabilities `json:"ioConnectivityCapabilities,omitempty"`
+
+	// IOPerformanceCapabilities for the storage pool
+	IOPerformanceCapabilities StoragePoolIOPerformanceCapabilities `json:"ioPerformanceCapabilities,omitempty"`
+
+	// DataProtectionCapabilities for the storage pool
+	DataProtectionCapabilities StoragePoolDataProtectionCapabilities `json:"dataProtectionCapabilities,omitempty"`
+}
+
+// StoragePoolReferenceResource
+type StoragePoolReferenceResource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+}
+
+// StoragePoolStorageCapacity for the pool,viz total, used and available capacity
+type StoragePoolStorageCapacity struct {
+	// TotalCapacity of the pool
+	TotalCapacity resource.Quantity `json:"totalCapacity,omitempty"`
+
+	// UsedCapacity of the pool
+	UsedCapacity resource.Quantity `json:"usedCapacity,omitempty"`
+
+	// AvailableCapacity of the pool
+	AvailableCapacity resource.Quantity `json:"availableCapacity,omitempty"`
+}
+
+// StoragePoolStorageIOPs for the pool,viz total, provisioned, used and available capacity
+type StoragePoolStorageIOPs struct {
+	// TotalIOPs of the pool
+	TotalIOPs uint64 `json:"totalIops,omitempty"`
+
+	// AvailableIOPs of the pool
+	AvailableIOPs uint64 `json:"availableIops,omitempty"`
+
+	// ProvisionedIOPs of the pool
+	ProvisionedIOPs uint64 `json:"provisionedIops,omitempty"`
+
+	// UsedIOPs of the pool
+	UsedIOPs uint64 `json:"usedIops,omitempty"`
+}
+
+type StoragePoolDeviceConfiguration struct {
+	// Devices for the list of UIDs of the devices
+	Devices []types.UID `json:"devices,omitempty"`
+
+	// RaidConfiguration for the raid configurations currently being used
+	RaidConfiguration map[RaidType]int `json:"raidConfiguration,omitempty"`
+}
+
+// StoragePoolState for the state of the pool for various scenarios
+type StoragePoolState string
+
+const (
```

Comment:
do we need these many state?

Reply:
We may not need these many states for our use case. I have listed all possible states that we use for cstor engine pools. 

Keeping it for now, we can remove the not needed states after further discussion.

Reply:
If we are using phase and conditions, then we don't need this state. Also, this can be specific to the pool, to be computed and filled by the pool provisioner plug-in.

Reply:
Just a thought as this pool CR is used in control plane can we put the status related to control plane or some common status?

Reply:
overall status? Like `healthy`, `degraded`, `unhealthy`, `unknown` ? That would be similar to conditions and phase.